predicators/envs/doors.py

```diff
@@ -729,3 +738,354 @@ def _get_open_door_target_value(mass: float, friction: float,
+            task_cache: Dict[str, Dict] = {
+                "static_geom": {},
+                "door_to_rooms": {},
+                "room_to_doors": {},
+                "door_to_doorway_geom": {},
+                "position_in_doorway": {},
+            }
+            state_with_cache = utils.StateWithCache(state.data, task_cache)
+            if not self.state_has_collision(state_with_cache):
+                return state_with_cache
+
+    @classmethod
+    def object_to_geom(cls,
+                       obj: Object,
+                       state: State,
+                       robot_pos: Optional[Array] = None) -> _Geom2D:
+        """Public for use by oracle options."""
+        if robot_pos is None:
+            x = state.get(obj, "x")
+            y = state.get(obj, "y")
+        else:
+            x, y = robot_pos
+        if obj.is_instance(cls._robot_type):
+            return utils.Circle(x, y, cls.robot_radius)
+        # Only the robot shape is dynamic. All other shapes are cached.
+        assert isinstance(state, StateWithCache)
+        static_geom_cache = state.cache["static_geom"]
+        if obj not in static_geom_cache:
+            if obj.is_instance(cls._room_type):
+                width = cls.room_size
+                height = cls.room_size
+                theta = 0.0
+            elif obj.is_instance(cls._door_type):
+                width = cls.hallway_width
+                height = cls.wall_depth
+                theta = state.get(obj, "theta")
+            elif obj.is_instance(cls._knob_type):
+                width = cls.hallway_width
+                height = cls.wall_depth
+                theta = state.get(obj, "theta")
+            else:
+                assert obj.is_instance(cls._obstacle_type)
+                width = state.get(obj, "width")
+                height = state.get(obj, "height")
+                theta = state.get(obj, "theta")
+            geom = Rectangle(x=x, y=y, width=width, height=height, theta=theta)
+            static_geom_cache[obj] = geom
+        return static_geom_cache[obj]
+
+    def _sample_door_feats(self, room_x: float, room_y: float, loc: str,
+                           rng: np.random.Generator) -> Dict[str, float]:
+        # This is the length of the wall on one side of the door.
+        offset = (self.room_size + self.wall_depth - self.hallway_width) / 2
+
+        if loc == "bottom":
+            x = room_x + offset
+            y = room_y - self.wall_depth / 2
+            theta = 0.0
+        else:
+            assert loc == "left"
+            x = room_x + self.wall_depth / 2
+            y = room_y + offset
+            theta = np.pi / 2
+
+        mass, friction, _ = rng.uniform(0.0, 1.0, size=(3, ))
+        return {
+            "x": x,
+            "y": y,
+            "theta": theta,
+            "mass": mass,
+            "friction": friction,
+            "open": 0.0,  # always start out closed
+        }
```

Comment:
this also looks identical to the superclass?

Reply:
I got rid of the original  "rot" and "target_rot" features. tbh I could also j add them back since it doesn't make much of a difference in the learning process